### PR TITLE
[Feature] Add basic BATCH api support

### PR DESCRIPTION
**Pull request by VishaalK**
1. We should rename `batchRequests` and `interface BatchRequest` to something more descriptive or indicative of what it does

### CODE REVIEW

File: target/oneNoteApi.js [hunk 1]
```diff
@@ -291,6 +291,31 @@ var OneNoteApi = (function (_super) {
         return this.requestPromise(url, form.asBlob(), form.getContentType());
     };
     /**
+     * BatchRequests
+     **/
+    OneNoteApi.prototype.batchRequests = function (batchRequests) {
+        this.enableBetaApi();
+        var boundaryName = "batch_" + Math.floor(Math.random() * 1000);
+        var contentType = "Content-Type: application/http";
```
Comment (mttwc) [inline target/oneNoteApi.js]: Remove

File: target/oneNoteApi.js [hunk 2]
```diff
@@ -291,6 +291,31 @@ var OneNoteApi = (function (_super) {
...
+            req += "\r\n";
+            req += batchRequest.httpMethod + " " + batchRequest.uri + " " + "HTTP/1.1" + "\r\n";
+            req += "Content-Type: " + batchRequest.contentType + "\r\n";
+            req += "\r\n";
+            req += batchRequest.content + "\r\n";
+            req += "\r\n";
+            data += req;
+        });
+        data += "--" + boundaryName + "--\r\n";
+        return this.requestBasePromise("/$batch", data, 'multipart/mixed; boundary="' + boundaryName + '"', "POST").then(this.disableBetaApi.bind(this));
```
Comment (mttwc) [inline target/oneNoteApi.js]: Race condition: Possible for another interleaving API call to hit beta api. Consider implementing enableBetaApiForNextRequest()

File: target/oneNoteApi.js [hunk 3]
```diff
@@ -418,6 +455,13 @@ var OneNoteApiBase = (function () {
         this.timeout = timeout;
         this.headers = headers;
     }
+    OneNoteApiBase.prototype.requestBasePromise = function (partialUrl, data, contentType, httpMethod) {
```
Comment (mttwc) [inline target/oneNoteApi.js:490]: use contentType = "application/json"

**Comment by VishaalK** [comment]
Talked with Matthew:

1. Changing interface of `batchRequests` to `BatchRequest` which is a class that we can add `operations` to.

The interface will then be `sendBatchRequest(request: BatchRequest)`, which makes it more clear that this function sends 1 HTTP request, and the object passed in is a collection of operations.

**Pull request merged**